sdk/docs/pages/typescript/executors.mdx

```diff
@@ -0,0 +1,119 @@
+executed through the executor.
+
+To fund transactions, the `SerialTransactionExecutor` will select all of the senders SUI coins for
+the first transaction, which will result in a single coin that will then be used as the gas payment
+on all subsequent transactions. This allows executing multiple transactions, without needing to
+re-query for gas coins, or wait for the RPC node to index the previous transactions.
+
+To further improve execution efficiency, the `SerialTransactionExecutor` caches the object versions
+of every object used or created by a transaction. This will significantly speed up the execution
+when multiple transactions use the same objects.
+
+`SerialTransactionExecutor` maintains an internal queue, so you don't need to wait for previous
+transactions to finish before sending the next one.
+
+```ts
+import { getFullnodeUrl, SuiClient } from '@mysten/sui/client';
+import { SerialTransactionExecutor } from '@mysten/sui/transactions';
+
+const client = new SuiClient({ url: getFullnodeUrl('devnet') });
+
+const executor = new SerialTransactionExecutor({
+	client,
+	signer: yourKeyPair,
+});
+
+const tx1 = new Transaction();
+const [coin1] = tx1.splitCoins(txb.gas, [1]);
+tx1.transferObjects([coin1], address1);
+const tx2 = new Transaction();
+const [coin2] = tx1.splitCoins(txb.gas, [1]);
+tx1.transferObjects([coin2], address2);
+
+const { digest: digest1 } = await executor.execute(tx1);
+const { digest: digest2 } = await executor.execute(tx2);
+```
+
+## `ParallelTransactionExecutor`
+
+The `ParallelTransactionExecutor` class works similarly to the `SerialTransactionExecutor`, but
+allows for parallel execution of transactions blocks. To make this work, the
+`ParallelTransactionExecutor` will maintain a pool of gas coins, and automatically execute
+additional transactions to refill the gas pool as needed.
+
+<Callout type="warning">
+	Using SuiClient or wallets to execute additional transactions while `ParallelTransactionExecutor`
+	is in use may consume/combine gas coins in the gasPool, causing transactions to fail.
+</Callout>
+
+In addition to managing gas and caching object versions, the `ParallelTransactionExecutor` will
+automatically detect what objects are being used by transactions, and schedules transactions in a
+way way that avoids conflicts between transactions using the same object ids.
+
+`ParallelTransactionExecutor` can be configured with a number of options:
+
+- `client`: An instance of `SuiClient` used to execute transactions.
+- `signer`: The signer/keypair used for signed transactions.
+- `coinBatchSize`: The maximum number of new coins to create when refilling the gas pool
+  (default 20)
+- `initialCoinBalance`: The balance of new coins created for the gas pool in MIST (default
+  `200_000_000n`),
+- `minimumCoinBalance`: After executing a transaction, the the gasCoin will be reused unless it's
+  balance is below this value (default `50_000_000n`),
```

Comment:
A lot of small transaction use a little less than 5_000_000 from what I've seen, I figured 10x that would be a reasonable minimum for to account for complex transactions.  Then 4x seems safe enough for an initial coin we hope to re-use a lot since most transactions will use less, that would be ~50 cheap transactions for 5 very expensive ones.

200_000_000 * 50 (max size) is 10 SUI which seems like a decent amount to fully fund the initial pool.

It's mostly completely made up without a lot of data to back it up 🤷